docs/source/processors/scratch.rst

```diff
@@ -0,0 +1,105 @@
+Scratch
+#######################################
+
+**Processor name:** ``scratch``
+
+You can include an image of Scratch blocks using
+`Scratch Block Plugin notation`_ using the following notation:
+
+.. literalinclude:: ../../../kordac/tests/assets/scratch/doc_example_basic_usage.md
+    :language: none
+
+to produce the following image:
+
+.. image:: ../images/scratch_blocks_example.png
+
+The syntax is the same for default Markdown code blocks, but Kordac handles the
+content differently due to the ``scratch`` language set at the start.
+
+You can test the output of your Scratch block text at
+`scratchblocks.github.io`_.
+
+.. warning::
+
+    Kordac doesn't create the Scratch images, but saves data for another system
+    (for example: Django) to create the images.
+    See :ref:`accessing-scratch-image-data` section below.
+
+The default HTML for button links is:
+
+.. literalinclude:: ../../../kordac/html-templates/scratch.html
+    :language: css+jinja
+
+Using the following example tag:
+
+.. literalinclude:: ../../../kordac/tests/assets/scratch/doc_example_basic_usage.md
+    :language: none
+
+The resulting HTML would be:
+
+.. literalinclude:: ../../../kordac/tests/assets/scratch/doc_example_basic_usage_expected.html
+    :language: html
+
+.. _accessing-scratch-image-data:
+
+Accessing Scratch image data
+***************************************
+
+When Kordac encounters a code block with the Scratch language (see example
+above), it doesn't not generate the image but saves the enclosed text and hash
+of the text in the ``required_files`` attribute under the key
+``scratch_images``.
+
+The following is an example of the result of required files after the parsing
+two Scratch blocks, where the ``scratch_images`` key points to a set of
+``namedtuple`` objects containing a ``hash`` (string) and ``text`` (string):
+
+.. code-block:: python
+
+    required_files = {
+        "scratch_images": [
+            ScratchImageMetaData(
+              hash="12203e8a9d57d5a3bdc384522acd150c",
```

Comment:
Probably should change the hashes used throughout the documentation to be SHA-256